File: abc/249a.c
#include <stdio.h>

int get_distance(int sec, int meeter, int rest, int x)
{
	int result = 0;
	int m_per_sec = 0;
	int loop_time  = 0;

	if (x <= sec)
		result = meeter * x;
	else
	{
		loop_time = x/(sec + rest);
		result = (sec*meeter) * loop_time;
	}
	return (result);
}

int main()
{
	int *input;
	int i = 0;
	int taka = 0;
	int aoki = 0;

	*input = sizeof(int*);
	while (i < 7)
	{
		input[i] = sizeof(int);
		scanf("%d", input[i]);
		i++;
	}
	taka = get_distance(input[0], input[1], input[2], input[6]);
	aoki = get_distance(input[3], input[4], input[5], input[6]);

	if (taka == aoki)
		printf("Draw");
	else if (taka < aoki)
		printf("Aoki");
	else
		printf("Takahashi");

	return 0;
}

File: abc_rust/src/249a.rs
use proconio::input;

fn get_distance(sec:i16, meeter:i16, rest:i16, x:i16) -> i16 {
    let result:i16 = if x < (sec + rest)
    {
        meeter * x
    }
    else
    {

        let left_distance = (x - (sec + rest))*meeter;
        meeter * sec + left_distance
    };
    return result;
}

fn main() {

    input! {
        a:i16, b:i16, c:i16, d:i16, e:i16, f:i16, x:i16,
    }

    let taka = get_distance(a,b,c,x);
    // println!("Taka:{}", taka);
    let aoki = get_distance(d, e, f, x);
    // println!("Aoki:{}", aoki);

    if taka == aoki
    {
        println!("Draw");
    }
    else if taka < aoki
    {
        println!("Aoki");
    }
    else {
        println!("Takahashi");
    }

}
